heartbeat/portblock

```diff
@@ -584,55 +1111,99 @@ IptablesValidateAll()
   return $OCF_SUCCESS
 }
 
-if
-  ( [ $# -ne 1 ] )
-then
+# Detect firewall tool - improved version
+detect_firewall_tool() {
```

Comment:
if this package is updated on a setup that up to now would have used iptables though nftables would've been present we might introduce behavioral change for the case that the iptables (wrapper I think) behaves differently. Upstream this might be Ok but as a courtesy to downstream adopters we might provide a simple way as to influence the logic so that it stays with iptables as long as the executable is available.
My suggestion would be 'auto' and 'safe-auto'  as that downstream people can set the default to 'safe-auto'.
For detect_firewall_tool this would mean a safe parameter that modifies the logic accordingly.